src/lib/swapper/swappers/ThorchainSwapper/utils/addL1ToLongtailPartsToMemo.test.ts

```diff
@@ -0,0 +1,130 @@
+import { btcChainId, dogeChainId, ethChainId, toAssetId } from '@shapeshiftoss/caip'
+import { describe, expect, it } from 'vitest'
+
+import { getMaxBytesLengthByChainId } from '../constants'
+import { addL1ToLongtailPartsToMemo } from './addL1ToLongtailPartsToMemo'
+
+const AGGREGATOR_ADDRESS = '0xd31f7e39afECEc4855fecc51b693F9A0Cec49fd2'
+const BIG_ADDRESS = '0x32DBc9Cf9E8FbCebE1e0a2ecF05Ed86Ca3096Cb632DBc9Cf9E8FbCebE1e'
+const REALLY_BIG_ADDRESS =
+  '0x32DBc9Cf9E8FbCebE1e0a2ecF05Ed86Ca3096Cb632DBc9Cf9E8FbCebE1e0a2ecF05Ed86Ca3096Cb632DBc9Cf9E8Fb1bE1e0a2ecF05Ed86Ca3096Cb632DBc9Cf9E8FbCebE1e0a2ecF05Ed86Ca3096Cb632DBc9Cf9E8FbCebE1e0a2ecF05Ed86Ca3096Cb6'
+
+const COLLIDING_FINAL_ASSETID = toAssetId({
+  chainId: ethChainId,
+  assetReference: '0xa0b86991c6218b36c1d19d4a2e9eb0ce36dF4741',
+  assetNamespace: 'erc20',
+})
+
+const FINAL_ASSET_CONTRACT_ASSETID = toAssetId({
+  chainId: ethChainId,
+  assetReference: '0x8a65ac0E23F31979db06Ec62Af62b432a6dF4741',
+  assetNamespace: 'erc20',
+})
+
+const THORCHAIN_ASSETIDS_ONE_COLLISION = [FINAL_ASSET_CONTRACT_ASSETID, COLLIDING_FINAL_ASSETID]
+
+const slippageBps = 100 // 1%
+
+describe('addL1ToLongtailPartsToMemo', () => {
+  it('should add aggregator address, shortened destination address and finalAssetAmountOut correctly', () => {
+    const finalAssetAmountOut = '9508759019'
+    const quotedMemo = `=:ETH.ETH:0x32DBc9Cf9E8FbCebE1e0a2ecF05Ed86Ca3096Cb6:42:ss:100`
+
+    const modifiedMemo = addL1ToLongtailPartsToMemo({
+      sellChainId: ethChainId,
+      quotedMemo,
+      aggregator: AGGREGATOR_ADDRESS,
+      finalAssetContractAssetId: FINAL_ASSET_CONTRACT_ASSETID,
+      finalAssetAmountOut,
+      slippageBps,
+      longtailTokens: THORCHAIN_ASSETIDS_ONE_COLLISION,
+    })
+
+    expect(modifiedMemo).toBe(
+      // The aggregator will turn the finalAssetAmountOut from 94136714201 to 9413671420 using the last 2 bytes as exponents
+      `=:e:0x32DBc9Cf9E8FbCebE1e0a2ecF05Ed86Ca3096Cb6:42:ss:100:d2:a6df4741:94136714201`,
+    )
+  })
+
+  it('should throw if chainId is BCH and memo length is > 220', () => {
+    const finalAssetAmountOut = '2083854765519275828179229'
+    const memoOver220Bytes = `=:ETH.ETH:${REALLY_BIG_ADDRESS}:42:ss:100`
+
+    expect(memoOver220Bytes.length).toBe(221)
+
+    expect(() =>
+      addL1ToLongtailPartsToMemo({
+        sellChainId: dogeChainId,
+        quotedMemo: memoOver220Bytes,
+        aggregator: AGGREGATOR_ADDRESS,
+        finalAssetContractAssetId: FINAL_ASSET_CONTRACT_ASSETID,
```

Comment:
Non-blocking, doesn't need to be tackled in this PR but feel free to if that's a simple LSP rename: now that we're using an `AssetId` instead of an address, the obvious implication is this is an asset and not an EOA address, so we can remove the `contract` terminology